### PR TITLE
Add check if player can destroy the block

### PR DESCRIPTION
**Pull request by dblsaiko**

### CODE REVIEW

File: src/main/java/mcmultipart/api/multipart/IMultipart.java [hunk 1]
```diff
@@ -263,6 +263,10 @@ public default void onPlantGrow(IPartInfo part, BlockPos source) {
         part.getState().getBlock().onPlantGrow(part.getState(), part.getPartWorld(), part.getPartPos(), source);
     }
 
+    public default boolean canPlayerDestroy(IPartInfo part, EntityPlayer player) {
+        return true;
```
Comment (asiekierka) [inline src/main/java/mcmultipart/api/multipart/IMultipart.java:267]: Couldn't this by default check, say, if getPlayerRelativeBlockHardness is non-negative? Unless that is checked elsewhere.

Comment (dblsaiko) [reply src/main/java/mcmultipart/api/multipart/IMultipart.java:267]: That would make it impossible to break those blocks in creative mode.

Comment (asiekierka) [reply src/main/java/mcmultipart/api/multipart/IMultipart.java:267]: Shouldn't there be no impossible-to-destroy parts in creative mode?

Comment (dblsaiko) [reply src/main/java/mcmultipart/api/multipart/IMultipart.java:267]: Exactly. (Additionally, if getPlayerRelativeBlockHardness is negative, it wouldn't be able to break in survival in the first place I think)

**Pull request merged**